README.md

```diff
@@ -1,81 +1,178 @@
+| Fast code generation                                                                                                                      | ✅                                                       | ✅                                               | ❌                                                      | ❌                                                         |                                                     ✅ |
+
+Perhaps the strongest feature of **Protobuf-ES** mentioned above is its generation of idiomatic JavaScript and TypeScript code.  It adopts the best features from the community generators. This means no more clunky getters and setters. You can now use things like the spread operator and make use of the same JavaScript semantics you've grown used to.
+
+For example, given a Protobuf file such as:
+
+```protobuf
+syntax="proto3";
+
+message User {
+  string first_name = 1;
+  string last_name = 2;
+  bool active = 3;
+  User manager = 4;
+  repeated string locations = 5;
+  map<string, string> projects = 6;
+}
 ```
 
+You can initialize your objects conveniently using the `new` operator or passing an initializer object to constructors:
+
+```typescript
+// Using new
+let user = new User();
+
+// Using an object in the constructor
+user = new User({
+  firstName: "Homer",
+  lastName: "Simpson",
+  manager: {  // you can simply pass an initializer object for this nested message field
+      firstName: "Montgomery",
+      lastName: "Burns"
+  },
+});
+```
+
+In addition, you can use direct property access and work with the same familiar JavaScript syntax you're used to.  You won't get confusing methods like `getLocationsList`, `setLocationsList`, `getProjectsMap`, and `clearProjectsMap`.  You won't have to access nested messages by doing things like `msg.getManager().getLocationsList()`.  For example:
+
+```typescript
+user.active = true;
+user.manager.firstName = "Monty";
+msg.locations = ["Springfield"];
+msg.projects = {
+   "SPP": "Springfield Power Plant"
+};
+```
+
+
 To learn more, have a look at a complete [code example](https://github.com/bufbuild/protobuf-es/tree/main/packages/protobuf-example), 
 the documentation for the [generated code](https://github.com/bufbuild/protobuf-es/blob/main/docs/generated_code.md), 
 and the documentation for the [runtime API](https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md).
 
+## Getting Started
 
-### How does this compare to protoc's JavaScript generator?
+Installation instructions vary depending on your purpose.  The instructions for npm are shown below but Yarn and pnpm are also supported.
 
-[`js_generator.cc`](https://github.com/protocolbuffers/protobuf-javascript/blob/main/generator/js_generator.cc)
-is rarely updated, and has fallen behind the quickly moving world of JavaScript.
+### [@bufbuild/protobuf](https://www.npmjs.com/package/@bufbuild/protobuf)
```

Comment:
I'm not sure this really gets new users started. Compare it to this [quickstart](https://github.com/timostamm/protobuf-ts#quickstart), which truly gets you generating code right away. 

It would be awesome if we would pick up the very simple `User` example from above, so that users have the full picture right away and can actually copy&paste and run the code.